libs/chain-signatures/contract/Cargo.toml

```diff
@@ -104,3 +104,7 @@ rstest = "0.25.0"
 test-utils = ["rand"]
 dev-utils = []
 abi = ["borsh/unstable__schema", "mpc-primitives/abi"]
+# This is used when running `cargo clippy --all-features`, because otherwise `abi` feat will break compilation.
+# But we can't put it under `abi` feat either, because it should appear only when building schema. In other words,
+# if it appears under wasm32 target, it will break compilation too.
+__abi-generate = ["near-sdk/__abi-generate"]
```

Comment:
Then something like:
`cargo clippy --features abi`
would fail. Isn't there a better solution?

Reply:
This is the best I got for the moment, it is not prettier, but it fixes somehow the issue above:
```
....
[target.'cfg(target_arch = "wasm32")'.dependencies]
getrandom = { version = "0.2.12", features = ["custom"] }
near-sdk = { version = "5.11.0", features = [
    "legacy",
    "unit-testing",
    "unstable",
] }

[target.'cfg(not(target_arch = "wasm32"))'.dependencies]
near-sdk = { version = "5.11.0", features = [
    "legacy",
    "unit-testing",
    "unstable",
    "__abi-generate",
] }
....
[features]
test-utils = ["rand"]
dev-utils = []
abi = ["borsh/unstable__schema", "mpc-primitives/abi"]
```
The bad part is that the `__abi-generate` is then enable for all non wasm32 targets by default. I am still searching for the most idiomatic fix for this situation.

Reply:
Looking at [near-sdk Cargo](https://github.com/near/near-sdk-rs/blob/master/near-sdk/Cargo.toml) I think they are using a similar trick, for example with package `wee_alloc`. It is a default feature but it only exists for the wasm32 target.

In a more general comment, I think we should avoid `schemars` and a few other dependencies for the wasm32 target, exactly as they do in `near-sdk-rs`, as that might help reducing contract size.

Reply:
> In a more general comment, I think we should avoid schemars and a few other dependencies for the wasm32 target, exactly as they do in near-sdk-rs, as that might help reducing contract size

Agree, I missed the part that it's till being linked, even though I don't enable introduces macros for wasm32

Reply:
It doesn't compile though, because if `__abi-generate` is enabled, then our `abi` should too. We can place `abi` under `default` feature for that

Reply:
I think our problem is mainly how `cargo-near` works. As it passes  the same features to both `cargo near abi` with default target, and `cargo near build` with wasm32 target. We don't have a clean way to say for the first use our abi feature, while dont use it for the second.

Reply:
If you are ok with generating abi by default, I think I got a fix
[diff.txt](https://github.com/user-attachments/files/21549829/diff.txt)
Basically I removed the abi flag from the contract and made everything conditional on the target wasm32, which includes which features are enabled for near-sdk. Therefore cargo-near has no issues, nor clippy --all-features with any target.

Reply:
`abi` and `near-sdk/__abi-generate` should be both enabled or disabled at the same time. But, when compiling a contract enabled `abi` doesn't lead to compilation failure... I don't understand what's happening

Reply:
another point to unify workspace dependencies...#702
I suggest leave it as it is (last successful CI, where we link the schemars into the contract). Let me double check how it affects the binary size

Reply:
ok, the difference is 1kb out of 953. I suggest we proceed we the changes, and get back to it later after #702

Reply:
follow up #759

Reply:
Just for the record, the reason why clippy failed on https://github.com/near/mpc/tree/gilcu3/fix_751 is that both legacy-mpc-contract and the current mpc-contract are used in `node` and `devnet`. So, the fixes that depend on target wasm32 that we were trying were failing in CI in the old contract, not the new one. That might mean that there is some type confusion somewhere in those two (node, devnet).